crates/fuel-core/src/service/adapters/producer.rs

```diff
@@ -85,3 +106,23 @@ impl fuel_core_producer::ports::Relayer for MaybeRelayerAdapter {
         }
     }
 }
+
+impl fuel_core_producer::ports::BlockProducerDatabase for Database {
+    fn get_block(&self, height: &BlockHeight) -> StorageResult<Cow<CompressedBlock>> {
+        let id = self.get_block_id(height)?.ok_or(not_found!("BlockId"))?;
+        self.storage::<FuelBlocks>()
+            .get(&id)?
+            .ok_or(not_found!(FuelBlocks))
+    }
+
+    fn block_header_merkle_root(&self, height: &BlockHeight) -> StorageResult<Bytes32> {
+        self.storage::<FuelBlockRoots>()
+            .get(height)?
+            .ok_or(not_found!(FuelBlocks))
```

Comment:
Also should be an option not an error

Reply:
The same as for `get_block`, we don't process an option on the caller's side